File: 0048-rotate-image/0048-rotate-image.java
class Solution {
    public void rotate(int[][] matrix) {
        int i = matrix.length;
        int j = matrix[0].length;
        int [][] arr=new int[i][j];
        for (int a=0;a<i;a++){
            for (int b=0;b<j;b++){
                arr[j-1-b][a]=matrix[a][b];
            }
        }
        for (int a=0;a<i;a++){
            for (int b=0;b<j;b++){
                matrix[i-1-a][j-1-b]=arr[a][b];}}
        
    }
}

File: 0048-rotate-image/0048-rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int rows = matrix.size();
        int cols = matrix[0].size();   
            vector<vector<int>> arr(rows, vector<int>(cols));

        for(int i=0;i<rows;i++){
            for(int j=0;j<cols;j++){
                arr[i][j]=matrix[i][j];
            }
        }
        for(int i=0;i<rows;i++){
            for(int j=0;j<cols;j++){
matrix[i][j] = arr[rows-1-j][i];            }
        }
    }
};